src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/AzureMetadataDeserializer.cs

```diff
@@ -20,18 +19,19 @@ public AzureMetadataDeserializer(ILogger<AzureMetadataDeserializer> logger) : ba
             Map(metadata => metadata.ResourceGroupName)
                 .IsRequired();
             Map(metadata => metadata.Cloud)
-                .WithDefault(AzureEnvironment.AzureGlobalCloud)
+                .WithDefault(AzureCloud.Global)
                 .MapUsing(DetermineAzureCloud);
         }
 
+        // TODO: validate cloud configuration in a SDK-agnostic way
         private object DetermineAzureCloud(string rawAzureCloud, KeyValuePair<YamlNode, YamlNode> nodePair, IErrorReporter errorReporter)
         {
             if (Enum.TryParse<AzureCloud>(rawAzureCloud, out var azureCloud))
             {
                 try
                 {
-                    var azureEnvironment = azureCloud.GetAzureEnvironment();
-                    return azureEnvironment;
+                    azureCloud.DetermineMetricsClientAudience();
+                    return azureCloud;
```

Comment:
Why do we call that specific method if we don't do anything with it?

Reply:
The previous flow here was to validate cloud support but also transform the field type from the SDK-agnostic `AzureCloud` to `AzureEnvironment`. Here I'm just performing the validation. I'm not using the transformed `Azure.Monitor` model to keep things SDK-agnostic